smarthash.py

```diff
@@ -191,35 +192,37 @@
     if len(nfos) > 0 and not imdb_id:
         nfo = nfos[0]
 
-    # manual imdb_id override
-    if args.imdb_id:
-        imdb_id = args.imdb_id
+    if 'imdb-id' in plugins[args.plugin].options:
+        # manual imdb_id override
+        if args.imdb_id:
+            imdb_id = args.imdb_id
 
-    # make sure the IMDb ID exists
-    if imdb_id and 'imdb-id' in plugins[args.plugin].options:
-
-        #imdb._logging.setLevel("error")
-        print('IMDb querying...\r', end='\r'),
-        imdb_site = imdb.IMDb()
-
-        imdb_movie = imdb_site.get_movie(imdb_id)
-        if not imdb_movie:
-            cprint("Invalid IMDb ID: {0}".format(imdb_id), "red")
-            sys.exit(1)
-        logging.info("IMDb verified: \"{0}\"".format(imdb_movie))
+        # make sure the IMDb ID exists
+        if imdb_id:
```

Comment:
This block of code currently won't run if there's an IMDb present which didn't come from a command line parameter

Reply:
I wasn't quite sure where else the id was meant to come from?  But as long as the plugin expects there to be an imdb-id this should work the same as it did before, I just changed where the check for the option was.  Was there a scenario where we expected to have an IMDb but the plugin wasn't expecting it?

Reply:
Lines 200-213 need to be unindented by one level.

Look at lines 185-189 - they scan text files in the directory to find an IMDb ID. If one is supplied using a command-line override, it'll take precedence. However, if one is not provided, validation of the ID and the call to choose_genre will not happen.